Add SubSlice to return slice of sub vipers

andig opened the pull request:
Replace #509. Fixes #171, #504, #800.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spf13/viper?pullRequest=863) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/spf13/viper?pullRequest=863) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spf13/viper?pullRequest=863) it.</sub>

andig commented:
@sagikazarmark I realize there is a big number of open PRs. Is this one you'd consider given it addresses a number of open issues?

andig commented:
~~ping~~ gentle nudge @sagikazarmark any chance to get your review?

andig commented:
I'm also watching https://github.com/spf13/viper/pull/866. In case that gets merged I'll check if it makes sense to carry the flag assignment over.

sagikazarmark commented:
Sorry @andig , my hands are full right now. I'll take a look at your PR when I get the chance,

criscola commented:
I would be interested in having this...

ben-turner commented:
I usually try to avoid "bump" and "+1" comments, but this seems like a fairly straight-forward PR that provides a lot of value, so I just wanted to bump it up to the top of the list before it gets lost to time.

andig commented:
Fixed the merge conflict while we're approaching 6th birthday of this PR. I can't even remember why I've added this. Is there still demand? Otherwise I'd like to close here ;)

andig commented:
Actually, all issues have been closed as "not planned". Closing here. Sorry for the noise.
